File: nasaPOD.py
import requests
import json
# import webbrowser

def nasaPOD(self, date):
	#API KEY
	API_KEY = 'E24iDq9jhvLGggYsFugUinAXI2aAeropVkq8DNZz'

	#API URL
	url='https://api.nasa.gov/planetary/apod'

	#Parameters
	params={
		'api_key' : API_KEY,
		'hd' : 'True',
		#'date' : '1997-03-22',
		'date': date
	}

	response = requests.get(url,params=params)
	json_data = json.loads(response.text)
	print(json_data)

	image_url= json_data['hdurl']
	# webbrowser.open(image_url)

	return image_url

File: NASA_POD.js
//API key
API_KEY = 'E24iDq9jhvLGggYsFugUinAXI2aAeropVkq8DNZz'

//API url
api_url = 'https://api.nasa.gov/planetary/apod'

function nasaPOD(date) {
    return fetch(api_url+`?api_key=${API_KEY}&hd=True&date=${date}`, {"SameSite":"Strict"}).then((response) => {
        //get json
        return response.json()
    }).then(json =>{
        if(json["code"] === undefined){
            return {"date": json["date"], "img_url": json["hdurl"]} //image existed, return img url
        } else {
            new_date = `${current_date.getFullYear()}` + date.slice(4)
            return fetch(api_url+`?api_key=DEMO_KEY&hd=True&date=${new_date}`).then((response) => { //image didn't exist take date from this year
                json2 = response.json()
                return {"date": json2["date"], "img_url": json2["hdurl"]}
            })
        }
    })
}